sw/device/silicon_creator/rom/e2e/rom_e2e_static_critical_test.c

```diff
@@ -0,0 +1,140 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdbool.h>
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_hmac.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/hmac_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+#include "sw/device/silicon_creator/lib/base/boot_measurements.h"
+#include "sw/device/silicon_creator/lib/base/sec_mmio.h"
+#include "sw/device/silicon_creator/lib/manifest_def.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+#define RUN_TEST(test_)                    \
+  LOG_INFO("Starting test " #test_ "..."); \
+  test_();                                 \
+  LOG_INFO("Finished test " #test_ ": ok.");
+
+#define CHECK_WORD_ALIGNED(addr_)                 \
+  CHECK((uintptr_t)addr_ % sizeof(uint32_t) == 0, \
```

Comment:
Where does the requirement for `manifest_start` and friends to be aligned come from? If they're meant to be arrays of `uint32_t`, I think we should use `alignof` here (and if that's the case, we could use `misalignment32_of()` from memory.h).

Reply:
It comes from the flash layout [here](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/lib/base/chip.h) and [here](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/rom/boot_policy_ptrs.h): flash is divided into two slots. Each slot, starts with a ROM_EXT image followed by a BL0 image. Both images are word aligned and manifest is at the start of these images (see [here](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/rom_ext/rom_ext_common.ld;drc=2c3bfae3e7ab3491aacffafc7e2dd2cdc6af4e0f;l=62)).